### PR TITLE
chore: update openzeppelin without interface changes

### PR DESCRIPTION
**Pull request by vrolland**

### CODE REVIEW

File: packages/smart-contracts/src/contracts/Bytes.sol [hunk 1]
```diff
@@ -10,7 +10,7 @@ library Bytes {
     * @notice Extract a bytes32 from a bytes.
     * @param data bytes from where the bytes32 will be extract
     * @param offset position of the first byte of the bytes32
-    * @return address
+    * @return bs
```
Comment (benjlevesque) [inline packages/smart-contracts/src/contracts/Bytes.sol:13]: what does bs mean? perfect place to describe it

File: packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol [hunk 2]
```diff
@@ -1,8 +1,7 @@
 // SPDX-License-Identifier: MIT
-pragma solidity >=0.4.25 <0.7.0;
+pragma solidity ^0.8.0;
 
-import "@openzeppelin/contracts/math/SafeMath.sol";
```
Comment (benjlevesque) [inline packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol:4]: Why this change? 
is math "safe" by default in newer solidity versions?

**Review by benjlevesque — APPROVED** [review]

File: packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol (hunk 2 above)
Comment (Dadogg80) [reply packages/smart-contracts/src/contracts/ChainlinkConversionPath.sol:4]: Hi. Yes, SafeMath is included from solidity version 0.8.4..

**Review by Dadogg80 — APPROVED** [review]
Looks good.

File: packages/smart-contracts/src/contracts/ERC20FeeProxy.sol [hunk 3]
```diff
@@ -17,12 +17,16 @@ contract ERC20FeeProxy {
   );
 
   // Fallback function returns funds to the sender
-  function() external payable {
+  fallback() external payable {
     revert("not payable fallback");
   }
```
Comment (yomarion) [inline packages/smart-contracts/src/contracts/ERC20FeeProxy.sol:22]: Do you think we still need a fallback ? My understanding is that since Solidity 0.6 if `receive()` is implemented and we try to call a function that does not exist, it will fail, hence I don't see the need for a "manual" revert.

Cf. here: https://blog.soliditylang.org/2020/03/26/fallback-receive-split/

Comment (yomarion) [reply packages/smart-contracts/src/contracts/ERC20FeeProxy.sol:22]: (I won't replicate the comment to all the other fallback implementations in other contracts FYI so if you agree, think about those)

File: packages/smart-contracts/src/contracts/ERC20SwapToPay.sol [hunk 4]
```diff
@@ -83,7 +84,9 @@ contract ERC20SwapToPay is Ownable {
 
     uint256 requestedTotalAmount = _amount + _feeAmount;
 
+    // solhint-disable-next-line
```
Comment (yomarion) [inline packages/smart-contracts/src/contracts/ERC20SwapToPay.sol:87]: Is this mandatory ?

File: packages/smart-contracts/src/contracts/StorageFeeCollector.sol [hunk 5]
```diff
@@ -1,7 +1,7 @@
-pragma solidity ^0.5.0;
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.0;
 
-import "@openzeppelin/contracts/math/SafeMath.sol";
-import "@openzeppelin/contracts/access/roles/WhitelistAdminRole.sol";
+import "./legacy_openzeppelin/contracts/access/roles/WhitelistAdminRole.sol";
```
Comment (yomarion) [inline packages/smart-contracts/src/contracts/StorageFeeCollector.sol:4]: Have you considered using [Roles](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/be5ed7364b93daccbb74a09e3f5ec1be6c458097/contracts/access/Roles.sol) ?

File: packages/smart-contracts/test/contracts/StorageFeeCollector.test.ts [hunk 6]
```diff
@@ -167,9 +167,7 @@ describe('contract: StorageFeeCollector', () => {
       );
       const contentSize = BigNumber.from(1000);
 
-      await expect(storageFeeCollector.getFeesAmount(contentSize)).to.be.revertedWith(
-        'SafeMath: multiplication overflow',
-      );
+      await expect(storageFeeCollector.getFeesAmount(contentSize)).to.be.reverted;
```
Comment (yomarion) [inline packages/smart-contracts/test/contracts/StorageFeeCollector.test.ts:170]: From my experience, this is risky, I found a few tests failing for the wrong reason previously. Can you instead replace with the expected error message ?

**Review by yomarion — CHANGES_REQUESTED** [review]
Well done, that's a lot of work for such a tiny figure change 😅 Only one comment is blocking, the others are more discussions to be sure.

File: packages/smart-contracts/src/contracts/StorageFeeCollector.sol (hunk 5 above)
Comment (vrolland) [reply packages/smart-contracts/src/contracts/StorageFeeCollector.sol:4]: Yes, It was my previous PR. I dashback on it change the ABI's.

File: packages/smart-contracts/test/contracts/StorageFeeCollector.test.ts (hunk 6 above)
Comment (vrolland) [reply packages/smart-contracts/test/contracts/StorageFeeCollector.test.ts:170]: The new safeMath mechanism does not give any message.

File: packages/smart-contracts/src/contracts/ERC20FeeProxy.sol (hunk 3 above)
Comment (vrolland) [reply packages/smart-contracts/src/contracts/ERC20FeeProxy.sol]: Good point! Thanks.

File: packages/smart-contracts/test/contracts/StorageFeeCollector.test.ts (hunk 6 above)
Comment (yomarion) [reply packages/smart-contracts/test/contracts/StorageFeeCollector.test.ts:170]: Maybe since safeMath is a default we can remove these low level tests ?

**Review by yomarion — APPROVED** [review]
Wonderful
